### DISCUSSION TITLE
Is supabase phone auth available with Dart language

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#2752] [Questions] [answered]
I am trying to use the new Supabase phone authentication with Dart language. I tried the below code, it just auto completes for email and password, dont see phone in there.

`final response = await client.auth.signUp({ phone: '+14100000000})`

If it works for phone, somebody please add some snippets here, or point me to some links that uses dart as the language for phone authentication.

**Comment by dshukertjr** [comment, marked as answer]
Hi @sumchans!

There is a PR to add this feature here https://github.com/supabase/gotrue-dart/pull/28.

**Reply by sumchans** [reply]
This is awesome. Super fast. Signing in with email is not working for me as of now.. I received the email with the magic link, clicking on it gives me the Internal server error. Would you be having any idea?
```
              Supabase supabaseClient = new Supabase();
              final client = supabaseClient.initialize();
              final response =
                  await client.auth.signUp('abcdef@gmail.com', '123456');
              if (response.error != null) {
                print('Error: ${response.error?.message}');
              } else {
                print(response.data);
              }
```

**Reply by sumchans** [reply]
Could this be a temporary error, or am I missing something?

**Reply by dshukertjr** [reply]
@sumchans 
Sorry to hear that email sign up is not working. Could I ask few questions to find the problem? 

- What version of `supabase_flutter` are you using?
- Which platform is this happening, iOS, Android or Web?
- Could you provide the error log of when the error happens?

If all goes good, both data and error should be null and you should receive a confirmation email when you run the following code. 
```dart
  final supabase = await Supabase.initialize(
    url: 'YOUR_SUPABASE_URL',
    anonKey: 'YOUR_ANNON_KEY',
  );
  final response = await supabase.client.auth.signUp('abcdef@gmail.com', '123456');
  print('data: ${response.data}');
  print('error: ${response.error}');
```

After clicking the link on your email, you should be taken back to the app, which you can detect `AuthState` class. You can read more about how to detect clicking confirmation links on the [Supabase Flutter Quick Start Guide](https://supabase.io/docs/guides/with-flutter)

**Reply by sumchans** [reply]
I am using supabase with Dart, not the supabase_flutter pack. Tried it on the ios simulator. I see the user created, but when I click on the magic link to verify the user, thats when I get the the below error -
`{"code":500,"msg":"Internal server error","error_id":"bf95ca5f-058d-4e41-b03b-482bef80cd5c"}`

**Reply by sumchans** [reply]
I am implementing my own UI for user authentication, that's the reason why I am using the supabase package.

**Reply by dshukertjr** [reply]
Thanks @sumchans for the details. 

I am having trouble reproducing your issue. Could I ask few more questions?

You mentioned you used this code
```dart
Supabase supabaseClient = new Supabase();
final client = supabaseClient.initialize();
final response =
    await client.auth.signUp('abcdef@gmail.com', '123456');
if (response.error != null) {
  print('Error: ${response.error?.message}');
} else {
  print(response.data);
}
```
Could you confirm that you got `null` printed to your debug console when you run the above code. 

Then you mentioned you got `{"code":500,"msg":"Internal server error","error_id":"bf95ca5f-058d-4e41-b03b-482bef80cd5c"}` after clicking the link on the email. It seems like you don't have any mechanism to bring the user back to the app, but  are you seeing this error on your browser's console or Flutter app's debug console? If you are seeing the error message on Flutter's console, how are you bringing the user back to the app? 

Also, if you are using Flutter, [supabase_flutter](https://pub.dev/packages/supabase_flutter) package might be easier to use. `supabase_flutter` is not a UI library. It is pretty much the same thing as [supabase](https://pub.dev/packages/supabase), but in addition will take care of the deep link required to bring the users back from magic link or third party OAuth authentications. In your case, you can bring back the users to the app after them confirming the email address!

**Reply by sumchans** [reply]
Hello Tyler,
I see the error message when clicking the link on the email on the browser and on the flutter console I see null printed. I don't have any mechanism to bring the user back into the app as of now. I am using the supabase package because being a beginner I find it is a lot less code and easier to use. 

Thanks

**Reply by sumchans** [reply]
Hello Tyler, If there is a Supabase package available for Dart, then why wouldn't it be possible to use instead of supabase_flutter. Regarding my current setup what is the step I am missing here. Also would I be able to do the new phone authentication with the supabase_flutter package.

**Reply by dshukertjr** [reply]
>I see the error message when clicking the link on the email on the browser and on the flutter console I see null printed. 

In this case, it is a server error and not the error on the dart client, so you can open an issue [here](https://github.com/supabase/gotrue) if it persists. 

>If there is a Supabase package available for Dart, then why wouldn't it be possible to use instead of supabase_flutter.

`supabase` package is a dart only package, and does not have Flutter as its dependencies. Because it does not depend on Flutter, it can be used with Angular-dart or server side dart as well. Because it does not depend on Flutter, it does not have any mechanism to automatically persist auth state, or does not have any mechanism of deep linking to bring the user back to the app after email confirmation. 

`supabase_flutter` packages uses `supabase` package internally, and it adds features like persisting auth state or deep linking. Since `supabase_flutter` depends on `supabase` package, it has all the features offered on `supabase` package including phone auth when it is added to `supabase` package.

**Reply by sumchans** [reply]
Oh wow, Just to clear the confusion, by using the supabase_flutter package, did you say that I can use the new phone authentication method as of now. If that is there now, then I dont have to create a ticket for the issue I am having now with the supabase with dart package. Please confirm.

Thanks Tyler, for the detailed explanation.

**Reply by dshukertjr** [reply]
No problem at all!

Actually, we just launched a new version of `supabase_flutter` that supports phone auth!
https://pub.dev/packages/supabase_flutter

**Reply by sumchans** [reply]
Wow awesome, thats all I was looking for. Thanks Tyler.

**Reply by sumchans** [reply]
How to mark this discussion as answered?

**Reply by dshukertjr** [reply]
My pleasure!

Is there a check mark next my first post?
https://docs.github.com/assets/images/help/discussions/comment-mark-as-answer-button.png

**Reply by sumchans** [reply]
Done
